### PR TITLE
Add built-in M4 UDF

### PR DESCRIPTION
**Pull request by LeiRui**
Readme about M4 is in `docs/UserGuide/Query-Data/Select-Expression.md` and `docs/zh/UserGuide/Query-Data/Select-Expression.md`.

### CODE REVIEW

**Comment by LeiRui** [comment]
I need to conserve the website link of `docs/UserGuide/UDF-Library/M4.md` for some time, so I don't remove that doc right now.

File: integration/src/test/java/org/apache/iotdb/db/integration/IoTDBUDFM4IT.java [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Locale;
+
+import static org.junit.Assert.fail;
+
+@Category({LocalStandaloneTest.class, ClusterTest.class})
+public class IoTDBUDFM4IT {
```
Comment (lancelly) [inline integration/src/test/java/org/apache/iotdb/db/integration/IoTDBUDFM4IT.java:43]: This test should be put in integration-test.

Comment (LeiRui) [reply integration/src/test/java/org/apache/iotdb/db/integration/IoTDBUDFM4IT.java]: Moved to integration-test.

Comment (lancelly) [reply integration/src/test/java/org/apache/iotdb/db/integration/IoTDBUDFM4IT.java]: Please merge master to see if the CI could pass.

Comment (LeiRui) [reply integration/src/test/java/org/apache/iotdb/db/integration/IoTDBUDFM4IT.java]: Seems that the master itself hasn't passed the CI yet.

**Review by qiaojialin — APPROVED** [review]

**Pull request merged**